examples/vision/perceiver_image_classification.py

```diff
@@ -0,0 +1,471 @@
+"""
+Title: Image Classification with Perceiver
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/04/30
+Last modified: 2021/01/30
+Description: Implementing the Perceiver model for image classification.
+"""
+
+"""
+## Introduction
+
+This example implements the [Perceiver: General Perception with Iterative Attention](https://arxiv.org/abs/2103.032069)
+model by Andrew Jaegle et al. for image classification,
+and demonstrates it on the CIFAR-100 dataset.
+The Perceiver model leverages an asymmetric attention mechanism to iteratively
```

Comment:
It would be useful to start with an overview of how it works, what problems it solves, and why it's potentially valuable / when to use it.

Reply:
Done

Reply:
Done